### ISSUE TITLE
Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by CatsMiaow** [discordjs/discord.js#3360]
- discord.js version: master https://github.com/discordjs/discord.js/commit/d7b2146c81d401d0fdefd8ec040a54dfce6067e5
- node.js version: v10.16.0

```js
/home/bot/node_modules/discord.js/src/stores/DataStore.js:20
    const existing = this.get(id || data.id);
                                         ^

TypeError: Cannot read property 'id' of undefined
    at Map.add (/home/bot/node_modules/discord.js/src/stores/DataStore.js:20:42)
    at ClientApplication._patch (/home/bot/node_modules/discord.js/src/structures/ClientApplication.js:73:83)
    at new ClientApplication (/home/bot/node_modules/discord.js/src/structures/ClientApplication.js:17:10)
    at Message._patch (/home/bot/node_modules/discord.js/src/structures/Message.js:152:43)
    at new Message (/home/bot/node_modules/discord.js/src/structures/Message.js:42:20)
    at Map.add (/home/bot/node_modules/discord.js/src/stores/DataStore.js:24:32)
    at Map.add (/home/bot/node_modules/discord.js/src/stores/MessageStore.js:18:18)
    at MessageCreateAction.handle (/home/bot/node_modules/discord.js/src/client/actions/MessageCreate.js:13:40)
    at Object.module.exports [as MESSAGE_CREATE] (/home/bot/node_modules/discord.js/src/client/websocket/handlers/MESSAGE_CREATE.js:4:32)
    at WebSocketManager.handlePacket (/home/bot/node_modules/discord.js/src/client/websocket/WebSocketManager.js:391:31)
    at WebSocketShard.onPacket (/home/bot/node_modules/discord.js/src/client/websocket/WebSocketShard.js:418:22)
    at WebSocketShard.onMessage (/home/bot/node_modules/discord.js/src/client/websocket/WebSocketShard.js:274:10)
    at WebSocket.onMessage (/home/bot/node_modules/ws/lib/event-target.js:120:16)
    at WebSocket.emit (events.js:198:13)
    at Receiver.receiverOnMessage (/home/bot/node_modules/ws/lib/websocket.js:789:20)
    at Receiver.emit (events.js:198:13)

// shard.on('death', (child) => console.warn(`Shard[${shardId}] Death`, `${child.pid}:${child.connected}:${child.killed}`));
Shard[2] Death 31934:false:false
```

**Comment by Monbrey** [comment]
Don't have a lot of to go off here except that `data` is undefined, and it seems to track back to defining the application owner/team. There's been a few more commits addressing this issue so recommend you update to the latest master then try to reproduce the issue.

**Comment by CatsMiaow** [comment]
ok, I will update from https://github.com/discordjs/discord.js/commit/d7b2146c81d401d0fdefd8ec040a54dfce6067e5 to https://github.com/discordjs/discord.js/commit/b65a4f05da40a45e33f78fc43c9b8172a170a3df and watch.

**Comment by SpaceEEC** [comment]
Duplicate of #3358, which should be fixed with the referenced commit there (1dd4c041e08068a8979a14cbdc5be95233634f2d).
